Development

Alex-sklx0 opened the pull request.

restrepo0510 committed:
Add basic structure (files and folders) with supporting content obout the project (fn and nfn requirements are missing).

restrepo0510 committed:
correction links in index.md

Alex-sklx0 committed:
Creation of navigation table in 'analysis' folder

Alex-sklx0 committed:
Creation of navigation table in analysis folder

restrepo0510 committed:
Creation of navigation table in 'analysis'

restrepo0510 committed:
Creation of navigation table in 'analysis'

restrepo0510 committed:
Creation of navigation table in 'analysis' 2

GloriaP-6 committed:
Creation and update of the first 9 functional requirements

restrepo0510 committed:
Creation and update about next 7 functional requirements

Alex-sklx0 committed:
add latest functional requirements and markdown

Mariajimenez16 committed:
Add README file for analysis-1

Mariajimenez16 committed:
add latest no functional requirements and markdown

restrepo0510 committed:
requirements be right

alvarezrozo committed:
Improve markdown sections in many files

alvarezrozo committed:
Merge pull request #2 from restrepo0510/feature/analysis-1

Add README and minor adjustments for feature/analysis-1

restrepo0510 committed:
Add story pointing

restrepo0510 committed:
Spacing correction

alvarezrozo committed:
Merge pull request #3 from restrepo0510/feature/story-pointing

Feature/story pointing

restrepo0510 committed:
MVP Requeriments :) UwU

restrepo0510 committed:
MVP Requeriments correction

restrepo0510 committed:
MVP Requeriments correction

alvarezrozo committed:
Merge pull request #5 from restrepo0510/feature/mvp

Add MPV

Alex-sklx0 committed:
Add basic structure of the design folder and design files

Alex-sklx0 committed:
Add UX content

Alex-sklx0 committed:
UX content correction

Mariajimenez16 committed:
add fe.md with framework comparison and final decision

Mariajimenez16 committed:
add be.md with backend options and final decision

Mariajimenez16 committed:
add be.md with backend options and final decision

Alex-sklx0 committed:
Add img folder to desing folder

Alex-sklx0 committed:
Add img folder to design folder

restrepo0510 committed:
ADD designs about uml diagrams

Alex-sklx0 committed:
UX Correction

Alex-sklx0 committed:
UX Correction

restrepo0510 committed:
ADD designs about uml diagrams CORRECTIONNNNNNN :)

restrepo0510 committed:
ADD designs about uml diagrams CORRECTIONNNNNNN :)

Alex-sklx0 committed:
Add index.md content in design folder

restrepo0510 committed:
ADD designs about uml diagrams CORRECTIONNNNNNN2 :)

Alex-sklx0 committed:
ux and img names correction

GloriaP-6 committed:
Add application UI images

Alex-sklx0 committed:
ux.md and doc/index.md corrections

GloriaP-6 committed:
DMR correction/ adding coordinates for the restaurants

Alex-sklx0 committed:
use cases diagram correction

GloriaP-6 committed:
UI images remaing

Alex-sklx0 committed:
use cases diagram correctio (again)

restrepo0510 committed:
ADD NEW IMAGES ABOUT SEQUENCE DIAGRAMS

restrepo0510 committed:
definitive sequence diagrams

alvarezrozo committed:
Merge pull request #7 from restrepo0510/feature/design-2

definitive sequence diagrams

Alex-sklx0 committed:
Add client and server folders, with their respective readme.md files

Alex-sklx0 committed:
feat(requirements): add RF-21 to RF-26 user profile, favorites, reviews and product management

Alex-sklx0 committed:
feat(frontend): create basic folder structure

The initial project structure is added to the client folder:
- src/screens/ for screens (Login, Home, etc.)
- src/components/ for reusable components
- App.tsx as the main entry point

This organization will serve as the basis for frontend development.

Alex-sklx0 committed:
feat(frontend): create basic folder structure

The initial project structure is added to the client folder:
- src/screens/ for screens (Login, Home, etc.)
- src/components/ for reusable components
- App.tsx as the main entry point

This organization will serve as the basis for frontend development.

Alex-sklx0 committed:
feat(auth): implement login screen UI

Complete design of login screen, including email/password form, decorations, an navigation to registration.

GloriaP-6 committed:
feat(ui): update register screen layout

Alex-sklx0 committed:
feat(core): implements the project's base structure and functional connection between frontend and backend

- Initial structure of the client (Expo) and server (NestJS with TypeORM) is configured.
- Authentication module with functional endpoints (, , ) is implemented.
- Connect the mobile frontend to the backend using  with a local URL.
- Establish integration with the PostgreSQL database for user persistence.
- Verify registration, login, and user listing in .

Alex-sklx0 committed:
refactor: implement best practices and refine styles across authentication components

- Apply TypeScript strict typing with interfaces and type definitions
- Implement comprehensive error handling with proper exception management
- Add input validation with user-friendly error messages
- Enhance security with password hashing and sensitive data protection
- Improve code documentation with JSDoc comments in English
- Refactor styles into unified design system with consistent spacing and colors
- Create reusable FormField component to reduce code duplication
- Optimize component structure with proper separation of concerns
- Add accessibility features with proper labels and hints
- Implement loading states and disabled form interactions during API calls
- Standardize naming conventions with descriptive English identifiers
- Remove redundant code and consolidate shared styles
- Improve responsive design with consistent form sizing across devices

Alex-sklx0 committed:
Merge pull request #8 from restrepo0510/feature/RF-01-User-login-and-registration

Feature/rf 01 user login and registration

Alex-sklx0 committed:
feat: implement location hook with proper error handling and permissions

- Create useLocation custom hook with TypeScript interfaces
- Add comprehensive permission handling for location access
- Implement loading states and error management
- Include automatic location request on component mount
- Provide manual refresh capability via requestLocation function
- Add balanced accuracy for better battery-performance tradeoff
- Include proper JSDoc documentation with usage examples
- Use useCallback for optimized function references
- Add console error logging for debugging purposes
- Follow React best practices with proper state management

Alex-sklx0 committed:
feat(restaurants): implement initial NearbyRestaurantsScreen with modern UI

- Add responsive grid layout with 2-column restaurant cards
- Implement TOP 1 featured restaurant section with trophy badge
- Add search bar with real-time filtering functionality
- Include floating bottom navigation bar with icons
- Display restaurant badges, categories, and distance indicators
- Integrate geolocation service for nearby restaurants
- Add loading and error states with user feedback

Pending: minor visual adjustments and layout refinement

Alex-sklx0 committed:
Merge pull request #9 from restrepo0510/feature/RF-06-User-geolocation

Alex-sklx0 committed:
feat(map): integrate restaurant fetching and user location display

- Refactored restaurantService to use native fetch with /restaurants and /restaurants/nearby endpoints
- Fixed API type definitions to match backend data (removed non-existent fields like category and address)
- Ensured latitude/longitude are parsed as numbers to prevent AIRMapMarker coordinate errors
- Added user location indicator (blue circle) on the map
- Cleaned up unused code and improved overall readability

Alex-sklx0 committed:
feat(ui): redesign TOP 1 card layout and fix navigation states

- Update TOP 1 card with horizontal image layout
- Add active indicator circle to bottom navigation
- Fix user location marker positioning on map rotation

restrepo0510 committed:
Add profile screen and posibility to edit it

Mariajimenez16 committed:
feat(auth): enhance navigation flow and add logout functionality

- Update main screen navigation routing and structure
- Implement logout feature with proper session management
- Fix various navigation-related bugs and inconsistencies
- Improve authentication state handling across screens
- Resolve minor UI issues and navigation errors

Mariajimenez16 committed:
feat(profile): implement functional user profile with real data

- Apply best practices to ProfileScreen and EditProfileScreen components
- Integrate real user data display from authentication context
- Implement functional profile editing with proper state management
- Add form validation and error handling for profile updates
- Improve code structure and maintainability following React patterns

Mariajimenez16 committed:
feat(products): implement complete product management system

- Add products module with entities, DTOs, and services
- Connect restaurant module with product relationships
- Configure AuthContext for product access control
- Create ProductService
- Implement ProductsScreen for product list
- Add product-restaurant relationship and data integrity

Mariajimenez16 committed:
feat(products): implement complete product management system

- Add products module with entities, DTOs, and services
- Connect restaurant module with product relationships
- Configure AuthContext for product access control
- Implement ProductsScreen for product list
- Add product-restaurant relationship and data integrity

alvarezrozo committed:
Add mvp story points

Mariajimenez16 committed:
docs: add comprehensive README for client folder

- Document frontend structure and purpose
- Add installation and setup instructions
- Include project standards and best practices
- Add useful resources and recommended versions
- Format with Markdown for better readability

Mariajimenez16 committed:
docs: add comprehensive README for client folder

- Document frontend structure and purpose
- Add installation and setup instructions
- Include project standards and best practices
- Add useful resources and recommended versions
- Format with Markdown for better readability, readme english and spanish

Mariajimenez16 committed:
fix(database): resolve entity relationships and foreign key constraints

- Fix User entity inconsistencies and missing role field
- Modify foreign key relationships between users, restaurants and products
- Add business profile creation for BUSINESS role users
- Implement product addition functionality with proper entity relationships
- Resolve circular dependencies and relation mapping errors
- Update entity definitions to ensure proper cascade operations
- Fix restaurant automatic creation for business users
- Improve data integrity across user-restaurant-product relationships

Mariajimenez16 committed:
fix(navigation): resolve home to maps and maps to products links

- Fix navigation link from HomeScreen to MapScreen
- Fix navigation link from MapScreen to products list
- Update route parameters and navigation handlers
- Ensure proper screen transitions between main features
- Resolve TypeScript path validation errors in navigation

Mariajimenez16 committed:
feat(business): implement product editing from business profile

- Add product editing functionality in business profile
- Fix bug in products list display and state management
- Implement proper product update flow with form validation
- Resolve data synchronization issues in products list
- Improve business profile product management experience

Mariajimenez16 committed:
docs: add comprehensive README files for client and server directories

- Add detailed client README with setup instructions and project structure
- Add comprehensive server README with API documentation and deployment guide
- Include installation steps, technical specifications, and development workflows
- Document environment setup and configuration requirements for both frontend and backend

Alex-sklx0 committed:
Merge pull request #10 from restrepo0510/feature/RF-02-View-nearby-restaurants

feat(map): integrate restaurant fetching and user location display

Alex-sklx0 committed:
Merge branch 'development' into feature/RF-26-Edit-a-product

Alex-sklx0 committed:
Merge pull request #15 from restrepo0510/feature/RF-26-Edit-a-product

Feature/rf 26 edit a product

Alex-sklx0 committed:
Merge branch 'development' into feature/RF-21-user-profile

Alex-sklx0 committed:
Merge pull request #11 from restrepo0510/feature/RF-21-user-profile

Add profile screen and posibility to edit it

alvarezrozo commented on client/README.md:
```diff
@@ -0,0 +1,88 @@
+# Carpeta client
```
Esto no debería estar en español

alvarezrozo commented on server/README.md:
```diff
@@ -0,0 +1,87 @@
+# Carpeta `server`
```
Esto no debería estar en español

Mariajimenez16 committed:
docs: update client and server README files

- Improve client README with detailed setup instructions and project structure
- Enhance server README with comprehensive API documentation and deployment guide
- Add installation steps, technical specifications, and development workflows
- Document environment configuration and requirements for both frontend and backend
- Fix formatting and improve readability of both documentation files

Alex-sklx0 committed:
docs: add missing requirements to MVP scope

- Include additional functional and non-functional requirements in MVP

Alex-sklx0 committed:
docs: add client and server routes to doc/index.md

Alex-sklx0 committed:
feat: add address geocoding functionality

- Implement geocoding service to convert addresses to latitude/longitude
- Update auth DTO to support location data from addresses
- Modify restaurant service with geocoding integration
- Create EditBusinessProfile component for business location management
- Extend auth types to include geocoding response handling

Alex-sklx0 committed:
refactor(auth): improve login and registration with best practices

- Implement self-documenting naming conventions
- Apply SRP with object-oriented programming patterns
- Enhance type safety with comprehensive TypeScript typing
- Add robust error handling and validation
- Document codebase with English comments and JSDoc

Alex-sklx0 committed:
fix(ui): improve error messages and color contrast for readability

- Enhance error message clarity and user guidance
- Optimize color contrast for better accessibility
- Improve visual hierarchy in form validation
- Standardize error states across authentication flows

Alex-sklx0 committed:
fix(ui): improve color contrast in formfield for readability

Alex-sklx0 committed:
feat(geocoding): enhance address to coordinates conversion

- Improve geocoding accuracy in restaurant service
- Optimize location handling in auth service
- Update restaurant DTO with enhanced geocoding fields
- Refine coordinate conversion logic and error handling

Alex-sklx0 committed:
fix(routing): correct profile and editprofile route names

Alex-sklx0 committed:
feat(business): add image picker for business profile

Alex-sklx0 committed:
Merge pull request #17 from restrepo0510/feature/RF-01-User-login-and-registration-correction

Feature/rf 01 user login and registration correction

Alex-sklx0 committed:
Merge pull request #18 from restrepo0510/feature/RF-02-View-nearby-restaurants-correction

Feature/rf 02 view nearby restaurants correction
